src/store/company.ts

```diff
@@ -0,0 +1,153 @@
+import type { Request } from '../types/request';
+import type { Company, SupervisoryAuthority } from '../types/company';
+import type { StoreSlice } from '../types/utility';
+import { RequestState } from './request';
+import { fetchCompanyDataBySlug } from '../Utility/companies';
+import { trackingFields, defaultFields, inferRequestLanguage } from '../Utility/requests';
+import type { GeneratorSpecificState, GeneratorState } from './generator';
+import { produce } from 'immer';
+import { SavedIdData } from '../DataType/SavedIdData';
+
+export interface CompanyState {
+    current_company?: Company | SupervisoryAuthority;
+    batch?: string[];
+    setCompanyBySlug: (slug: string) => Promise<void>;
+    setCompany: (company: Company) => Promise<void>;
+    setSva: (sva: SupervisoryAuthority) => Promise<void>;
+    removeCompany: () => Promise<void>;
+    startBatch: (batch: string[]) => Promise<void>;
+    advanceBatch: () => Promise<void>;
+    clearBatch: () => void;
+    hasBatch: () => boolean | undefined;
+}
+
+export const createCompanyStore: StoreSlice<CompanyState, RequestState<Request> & GeneratorSpecificState> = (
+    set,
+    get
+) => ({
+    setCompany: async (company) => {
+        set(
+            produce((state: GeneratorState) => {
+                if (company) {
+                    state.current_company = company;
+                    state.request.language = inferRequestLanguage(company);
+                    state.request.slug = company.slug;
+                    state.request.recipient_runs = company.runs || [];
```

Comment:
Yes and no. I mostly left them to be able to keep some of the legacy code. And since we currently don't differentiate between SVAs and companies in the store there is no difference. I would still suggest to keep it for now to keep it simple, because the changes required to refactor this are bigger than you might think.